app/controllers/kid_account_shares_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class KidAccountSharesController < ApplicationController
+  before_action :authenticate_user!
+
+  def new; end
+
+  def create
+    parental_key = generate_unique_key
+    ActiveRecord::Base.transaction do
+      user = User.includes(:account).find_by(parental_key: account.parental_key) if account.parental_key?
```

Comment:
WDYT if we just create proper account share here? In other words, could this logic about finding the proper user or creating a new one be moved into the "kid login" step?